webapp/channels/src/components/widgets/modals/components/section_item_creator.tsx

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useIntl} from 'react-intl';
+
+import './section_item_creator.scss';
+
+export type SectionItemProps = {
+    title?: {
+        id: string;
+        defaultMessage: string;
+    };
+    description?: {
+        id: string;
+        defaultMessage: string;
+    };
+};
+
+type Props = SectionItemProps & {
+    content: JSX.Element;
+}
+
+function SectionItemCreator({title, description, content}: Props): JSX.Element {
```

Comment:
And for my last name nitpick, could you rename this to `BaseSettingItem`? I think that'd help differentiate it a bit more since I got `SectionCreator` and `SectionItemCreator` confused when reading through this at first